pkg/minikube/service/service.go

```diff
@@ -378,3 +378,26 @@ func DeleteSecret(cname string, namespace, name string) error {
 
 	return nil
 }
+
+func CheckServicePods(clientset typed_core.CoreV1Interface, svcName, namespace string) error {
+	svc, err := clientset.Services(namespace).Get(context.Background(), svcName, meta.GetOptions{})
+	if err != nil {
+		return err
+	}
+	if svc.Spec.Type != core.ServiceTypeNodePort && svc.Spec.Type != core.ServiceTypeLoadBalancer {
+		return nil
+	}
+	pods, err := clientset.Pods(namespace).List(context.Background(), meta.ListOptions{
+		LabelSelector: labels.Set(svc.Spec.Selector).AsSelector().String(),
+	})
+	if err != nil {
+		return err
+	}
+
```

Comment:
When there is a deployment has created these `Pods`, `restartPolicy` of the `Deployment` set to `Always` so the `pod.Status.Phase` would return `Running`, even in the case of there is a problem with `Pod` and it would still give URL and try to open web-browser.

So maybe checking the `ContainerStatuses` and readiness state might be helpful?

```
for _, containerStatus := range pod.Status.ContainerStatuses {
			if containerStatus.Ready == true {
				return nil
			}
		}
```